### PR TITLE
stop allow_guessing for command line option parser

### LINKED ISSUE #2148: CLI: should complain on unknown parameters
It silently ignores an unknown parameter such as `--allow-path test` or even `--allow-path=test`.

### PR DESCRIPTION
**Pull request by izgzhen**
this fixes https://github.com/ethereum/solidity/issues/2148

### CODE REVIEW

**Comment by chriseth** [comment]
Thanks for the pull request! Could you add a short line to `Changelog.txt` and perhaps also a test to `test/cmdlineTests.sh`?

**Review by chriseth — CHANGES_REQUESTED** [review]
Please add changelog entry and tests.

**Comment by izgzhen** [comment]
@chriseth thanks, will take care of it soon :)

**Comment by izgzhen** [comment]
updated

**Review by chriseth — APPROVED** [review]

**Comment by chriseth** [comment]
Thanks a lot!

File: test/cmdlineTests.sh [hunk 1]
```diff
@@ -83,6 +83,21 @@ function compileWithoutWarning()
...
+    set +e
+    output=$("$SOLC" --allow=test 2>&1)
+    failed=$?
+    set -e
+
+    if [ "$output" == "unrecognised option '--allow=test'" ] && [ $failed -ne 0 ] ; then
+	echo "Passed"
+    else
+	printError "Incorrect response to unknown options: $STDERR"
+	exit 1
```
Comment (axic) [inline test/cmdlineTests.sh:97]: Should adjust indentation here.

Comment (izgzhen) [reply test/cmdlineTests.sh:97]: oh, sorry

**Pull request merged**